alacritty/src/renderer/rects.rs

```diff
@@ -190,3 +195,216 @@ impl RenderLines {
         }
     }
 }
+
+/// Shader sources for rect rendering program.
+pub static RECT_SHADER_F_PATH: &str = concat!(env!("CARGO_MANIFEST_DIR"), "/res/rect.f.glsl");
+pub static RECT_SHADER_V_PATH: &str = concat!(env!("CARGO_MANIFEST_DIR"), "/res/rect.v.glsl");
+static RECT_SHADER_F: &str = include_str!("../../res/rect.f.glsl");
+static RECT_SHADER_V: &str = include_str!("../../res/rect.v.glsl");
+
+#[repr(C)]
+#[derive(Debug, Clone, Copy)]
+struct Rgba {
+    r: u8,
+    g: u8,
+    b: u8,
+    a: u8,
+}
+
+/// Struct that stores vertex 2D coordinates and color for rect rendering.
+#[repr(C)]
+#[derive(Debug, Clone, Copy)]
+struct Vertex {
+    // Normalized screen coordinates.
+    // TODO these can certainly be i16.
+    x: f32,
+    y: f32,
+
+    // Color.
+    color: Rgba,
+}
+
+/// Struct to group together rect-related GL objects and rendering functionality.
```

Comment:
I find this comments starting with `struct` a bit weird to read tbqh, it's not really a C, since we have `impl` blocks for the sturcts, etc. I think it'd be nice to not mention `struct` at all, since it's already marked as such in docs. This applies to other struct you have here (Can go without comment for it tbqh).

Reply:
Removed comments